### PR TITLE
Open up some java APIs for appending Strings, and fix some minor stiring issues

### PR DESCRIPTION
**Pull request by revans2**
This opens up java APIs for building a string or string category vector.  It fixes some issues where someone might not know the final size of a string buffer, and also fixes an issue with filter where it requires a string category not a string.  If you use a string the native code will abort the process.

### CODE REVIEW

File: java/src/main/java/ai/rapids/cudf/ColumnVector.java [hunk 1]
```diff
@@ -1997,71 +1998,87 @@ public final Builder append(double value) {
...
       assert currentIndex < rows;
-      assert (currentStringByteIndex + length) <= stringBufferSize;
+      // just for strings we want to throw a real exception if we would overrun the buffer
+      while (currentStringByteIndex + length > data.getLength()) {
+        long oldlen = data.getLength();
+        long newlen = oldlen * 2;
+        // need to grow the size of the buffer.
+        HostMemoryBuffer newData = HostMemoryBuffer.allocate(newlen);
+        try {
+          newData.copyFromHostBuffer(0, data, 0, currentStringByteIndex);
```
Comment (jlowe) [inline java/src/main/java/ai/rapids/cudf/ColumnVector.java]: Why allocate and copy all the data when we might have to do this again if the `while` loop doesn't terminate?  Ideally this should only allocate and copy once.

Comment (revans2) [reply java/src/main/java/ai/rapids/cudf/ColumnVector.java]: You are right, and there is a corner case here that I have to address.

**Comment by revans2** [comment]
@jlowe please take another look.

**Comment by jlowe** [comment]
No intention of adding a unit test as part of this?  Would be nice to have one where someone initializes a vector with a string buffer of 0 and appends at least one string to it.

**Comment by jlowe** [comment]
rerun tests

**Pull request merged**